lib/api/api.js

```diff
@@ -606,6 +614,44 @@ api.prototype.sendTransfer = function(seed, depth, minWeightMagnitude, transfers
     })
 }
 
+/**
+* Promotes a transaction by adding spam on top of it.
+* Will promote {maximum} transfers on top of the current one with {delay} interval.
+*
+* @Param {string} tail
+* @param {int} depth
+* @param {int} minWeightMagnitude
+* @param {array} transfer
+* @param {object} params
+* @param callback
+*
+    * @returns 
+*/
+api.prototype.promoteTransaction = function(tail, depth, minWeightMagnitude, transfer, params, callback) {
+    var self = this;
+    if (!inputValidator.isHash(tail)) {
+        return callback(errors.invalidTrytes());
+    }
+
+    if (!self.isPromotable(tail)) {
```

Comment:
I think we should get the resolved value here, like this: `self.isPromotable.then(isPromotable => { .... })`